pulp_rpm/app/shared_utils.py

```diff
@@ -9,6 +11,17 @@
 from django.utils.dateparse import parse_datetime
 
 
+def compressed_file_open(download_result):
```

Comment:
Should we go with something more like

```python
with tempfile.TemporaryDirectory(dir=".") as d:
    fname = .... # strip off compression suffix
    cr.decompress_file(path, os.path.join(d, fname), cr.detect_compression(path))

```

Downsides: less straightforwards, requires creating a new file and then reading it again
Upsides: probably more robust, adds zstd support for free

Reply:
I'm not sure that would work as written, we're creating new files that the code using this context manager doesn't know about.  But in general principle

Reply:
I'm not sure the complication makes it worth the candle, as it were. We'd have to change `detect_compression()` the same way we would `compressed_file_open()` in the event of new types, and it "feels" like manipulating extra files is asking for problems.

Reply:
As it turns out we pretty much have to anyway, because in the event that the comps file was uploaded, the file extension gets erased by the storage backend.  It might have been .gz or .xz or whatever and in that case we need to use a magic-number aware function.

Alternatively, I suppose we could save the temporary file uncompressed.  But then upload-in-parts might be problematic...

Reply:
Ah, very very good point. Relying on the filename is always problematic, let's do it "the right way".

RE comps - comps.xml isn't an artifact, we only store the various kinds-of comps stanzas as Content, and recreate comps.xml at publish-time.

Reply:
What I mean is - temporarily, during the comps.xml upload and content creation process, it's stored as a pulptemporaryfile which goes into the storage backend.  Do you think it's better to lazily decompress or eagerly decompress before it gets saved as a temporary file?

Reply:
I think we should leave it as-is, and decompress when necessary in the task-code. Eagerly-decompress at upload puts the decompress-time on the human waiting for the REST call to return; the current approach adds to the task-time instead of the human-wait-time. Does that make sense?